### PR TITLE
Mock Test -> Ginkgo Test Capabilities

### PR DESCRIPTION
**Pull request by zachmandeville**
This PR updates our mock-template slightly to work with new apisnoop org-mode capabilities, as documented in the tickets/k8s/.dir-locals.el  Namely, the ability to convert your mock test into one (mostly) ready for inclusion in the ginkgo framework.  

The functions isolates just the test from your mockup, then scans it for 3 different types of error messages, converting them to thee analogous gingko assertion and inserting it into the org file beneath the heading 'Ginkgo Test'.

This work makes use of named src blocks, a new 'ginkgo test' heading with a specific id and for the mock test to include comments `// TEST BEGINS HERE` and `// TEST ENDS HERE`.  Because of this, the functions may not work on tickets made without the mock template.

It is also advised to double-check the conversion before submitting yr ginkgo test as some adjustments may still be needed.  This function is meant mostly as a convenience, and not authoratative conversion.

### CODE REVIEW

File: tickets/k8s/.dir-locals.el [hunk 1]
```diff
@@ -1,17 +1,111 @@
...
-       (eval .
-             (defun apisnoop/delete-live-events ()
-               "deletes all live events from audit_events table using psql.  Requires you to have already
+           (
+            (eval . (setq-local RESOURCENAME "RESOURCENAME"))
+            (eval .
+                  (defun apisnoop/insert-mock-template ()
+                    "Inserts contents of current directory's mock-template.org file into current buffer."
+                    (interactive)
+                    (let ((mock-template "./mock-template.org"))
```
Comment (hh) [inline tickets/k8s/.dir-locals.el:8]: Might be good to detect IF we are in a org file and if not offer to create one in this folder.

File: tickets/k8s/.dir-locals.el [hunk 2]
```diff
@@ -1,17 +1,111 @@
...
+                      (let* ((mock-test (apisnoop/get-mock-test-value)) 
+                             (ginkgo-test (-> mock-test
+                                              apisnoop/format-nil-errors
+                                              apisnoop/format-eq-errors
+                                              apisnoop/format-comp-errors))
+                             (src-code-block (concat "#+NAME: Ginkgo Test\n"
+                                                     "#+begin_src go\n"
+                                                     (s-trim-right ginkgo-test)
+                                                     "\n#+end_src\n")))
+                        (goto-char (org-find-entry-with-id "gt001z4ch1sc00l"))
```
Comment (hh) [inline tickets/k8s/.dir-locals.el:106]: Is there a hidden message in the id? 🥇

**Comment by BobyMCbobs** [comment]
This is really good! Great work @zachmandeville 

/approve

**Pull request merged**